src/libraries/Common/src/System/Security/Cryptography/RSACng.ImportExport.cs

```diff
@@ -64,26 +64,7 @@ public override unsafe void ImportParameters(RSAParameters parameters)
                 {
                     throw new CryptographicException(SR.Cryptography_InvalidRsaParameters);
                 }
-
-                // Half, rounded up.
-                int halfModulusLength = (parameters.Modulus.Length + 1) / 2;
-
-                // The same checks are done by RSACryptoServiceProvider on import (when building the key blob)
-                // Historically RSACng let CNG handle this (reporting NTE_NOT_SUPPORTED), but on RS1 CNG let the
```

Comment:
@bartonjs this comment was a bit concerning to me. Though I have a test saying otherwise, this comment appears to be saying that CNG, as recently as RS1, this was not expected to work, but the test in `SignVerify.cs` appears to pass.

Reply:
Returning to corefx for the history I see this was PR https://github.com/dotnet/corefx/pull/39087 and issue (post-move) https://github.com/dotnet/runtime/issues/29636.

https://github.com/dotnet/runtime/blob/5183c6e0e8e1ef726f6c61b0d8bb56fb255b698f/src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/RSAXml.cs#L1342-L1381

The parameters are clearly invalid, as P.Length + Q.Length isn't Modulus.Length +/- 1.  (Even if leading zeroes were trimmed off).

Similar arguments could be made if DP.Length > P.Length, DQ.Length > Q.Length, or QInv.Length > P.Length (since it's (1/Q % P))

Reply:
I don't suppose you got ahold of a Win10-RS1 machine to run the tests and see what fails here?

Reply:
@bartonjs I think I can manage to spin up a VM.

Reply:
Mkay, I was able to cobble together a Windows 10 2016 LTSB VM and indeed that test fails, no exception is thrown right now. Will address.